lib/spack/spack/test/concretize_preferences.py

```diff
@@ -84,6 +85,16 @@ def test_preferred_variants(self):
             'mpileaks', debug=True, opt=True, shared=False, static=False
         )
 
+    def test_preferred_variants_from_any(self):
```

Comment:
Can you add a docstring to this test? Something like "when a variant default is not "none", the user can specify foo=any on the command line, but when it is "none" they cannot (since "none" does not satisfy "any").